### PR TITLE
Use pytest-xdist for parallel execution of tests

### PR DESCRIPTION
**Pull request by rogalski**

### CODE REVIEW

**Comment by rogalski** [comment]
Maybe there should be upper bound for -n parameter, since on travis, 32 processes are used.

```
============================= test session starts ==============================
platform linux2 -- Python 2.7.9, pytest-3.0.5, py-1.4.32, pluggy-0.4.0
rootdir: /home/travis/build/PyCQA/pylint, inifile: pytest.ini
plugins: xdist-1.15.0
gw0 [811] / gw1 [811] / gw2 [811] / gw3 [811] / gw4 [811] / gw5 [811] / gw6 [811] / gw7 [811] / gw8 [811] / gw9 [811] / gw10 [811] / gw11 [811] / gw12 [811] / gw13 [811] / gw14 [811] / gw15 [811] / gw16 [811] / gw17 [811] / gw18 [811] / gw19 [811] / gw20 [811] / gw21 [811] / gw22 [811] / gw23 [811] / gw24 [811] / gw25 [811] / gw26 [811] / gw27 [811] / gw28 [811] / gw29 [811] / gw30 [811] / gw31 [811]
scheduling tests via LoadScheduling

```

File: tox.ini [hunk 1]
```diff
@@ -16,12 +16,13 @@ deps =
...
    mccabe
    pytest
+   pytest-xdist
 
 setenv =
     COVERAGE_FILE = {toxinidir}/.coverage.{envname}
 
 commands =
-    python -Wi {envsitepackagesdir}/coverage run -m pytest {envsitepackagesdir}/pylint/test/ {posargs:}
+    python -Wi {envsitepackagesdir}/coverage run -m pytest -n auto {envsitepackagesdir}/pylint/test/ {posargs:}
```
Comment (PCManticore) [inline tox.ini]: I would prefer to keep ``tox.ini`` untouched, but to provide the flag to pytest through ``posargs``, in ``travis.yml``.

Comment (rogalski) [inline tox.ini]: Is there any good reason to keep this feature disabled by default? Manual runs by developers and contributors and done much more often than automated Travis runs.

Comment (PCManticore) [inline tox.ini]: If contributors and developers want, they can pass the flags themselves to tox. It does not have to be a "one size fits it all" entry point for running the tests.

**Pull request merged**